precompiles/erc20/setup_test.go

```diff
@@ -0,0 +1,72 @@
+package erc20_test
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	erc20precompile "github.com/evmos/evmos/v15/precompiles/erc20"
+	"github.com/evmos/evmos/v15/testutil/integration/evmos/factory"
+	"github.com/evmos/evmos/v15/testutil/integration/evmos/grpc"
+	testkeyring "github.com/evmos/evmos/v15/testutil/integration/evmos/keyring"
+	"github.com/evmos/evmos/v15/testutil/integration/evmos/network"
+	utiltx "github.com/evmos/evmos/v15/testutil/tx"
+	erc20types "github.com/evmos/evmos/v15/x/erc20/types"
+	"github.com/evmos/evmos/v15/x/evm/statedb"
+	"github.com/stretchr/testify/suite"
+)
+
+var s *PrecompileTestSuite
+
+// PrecompileTestSuite is the implementation of the TestSuite interface for ERC20 precompile
+// unit tests.
+type PrecompileTestSuite struct {
+	suite.Suite
+
+	network     network.Network
+	factory     factory.TxFactory
+	grpcHandler grpc.Handler
+	keyring     testkeyring.Keyring
+
+	stateDB *statedb.StateDB
+
+	precompile *erc20precompile.Precompile
+}
+
+func TestPrecompileTestSuite(t *testing.T) {
+	s = new(PrecompileTestSuite)
+	suite.Run(t, s)
+}
+
+func (s *PrecompileTestSuite) SetupTest() {
+	keyring := testkeyring.New(1)
+	integrationNetwork := network.NewUnitTestNetwork(
+		network.WithPreFundedAccounts(keyring.GetAllAccAddrs()...),
+	)
+	grpcHandler := grpc.NewIntegrationHandler(integrationNetwork)
+	txFactory := factory.New(integrationNetwork, grpcHandler)
+
+	headerHash := integrationNetwork.GetContext().HeaderHash()
+	stateDB := statedb.New(
```

Comment:
You dont need this anymore, check the other PR where you had done the comment. I added a `GetStateDB` method for UnitTestNetwork`

Reply:
awesome!

Reply:
I saw that this is on a PR that's not yet merged - I will be adding more unit tests so I'll adjust to use the util on a follow-up PR 👍